src/messaging/tests/MessagingContext.h

```diff
@@ -166,135 +163,155 @@ class MessagingContext : public PlatformMemoryUser
     CHIP_ERROR CreateAliceFabric();
     CHIP_ERROR CreateBobFabric();
 
-    const Transport::PeerAddress & GetAliceAddress() { return mAliceAddress; }
-    const Transport::PeerAddress & GetBobAddress() { return mBobAddress; }
+    const Transport::PeerAddress & GetAliceAddress() { return mpData->mAliceAddress; }
+    const Transport::PeerAddress & GetBobAddress() { return mpData->mBobAddress; }
 
     Messaging::ExchangeContext * NewUnauthenticatedExchangeToAlice(Messaging::ExchangeDelegate * delegate);
     Messaging::ExchangeContext * NewUnauthenticatedExchangeToBob(Messaging::ExchangeDelegate * delegate);
 
     Messaging::ExchangeContext * NewExchangeToAlice(Messaging::ExchangeDelegate * delegate, bool isInitiator = true);
     Messaging::ExchangeContext * NewExchangeToBob(Messaging::ExchangeDelegate * delegate, bool isInitiator = true);
 
-    System::Layer & GetSystemLayer() { return mIOContext->GetSystemLayer(); }
+    System::Layer & GetSystemLayer() { return mpData->mIOContext->GetSystemLayer(); }
 
 private:
-    bool mInitializeNodes = true;
-    bool mInitialized;
-    FabricTable mFabricTable;
-
-    SessionManager mSessionManager;
-    Messaging::ExchangeManager mExchangeManager;
-    secure_channel::MessageCounterManager mMessageCounterManager;
-    IOContext * mIOContext;
-    TransportMgrBase * mTransport;                // Only needed for InitFromExisting.
-    chip::TestPersistentStorageDelegate mStorage; // for SessionManagerInit
-    chip::PersistentStorageOperationalKeystore mOpKeyStore;
-    chip::Credentials::PersistentStorageOpCertStore mOpCertStore;
-    chip::Crypto::DefaultSessionKeystore mSessionKeystore;
-
-    FabricIndex mAliceFabricIndex = kUndefinedFabricIndex;
-    FabricIndex mBobFabricIndex   = kUndefinedFabricIndex;
-    GroupId mFriendsGroupId       = 0x0101;
-    Transport::PeerAddress mAliceAddress;
-    Transport::PeerAddress mBobAddress;
-    Transport::PeerAddress mCharlieAddress;
-    Transport::PeerAddress mDavidAddress;
-    SessionHolder mSessionAliceToBob;
-    SessionHolder mSessionBobToAlice;
-    SessionHolder mSessionCharlieToDavid;
-    SessionHolder mSessionDavidToCharlie;
-    Optional<Transport::OutgoingGroupSession> mSessionBobToFriends;
+    struct MessagingContextData
+    {
+        MessagingContextData() :
+            mInitialized(false), mAliceAddress(Transport::PeerAddress::UDP(GetAddress(), CHIP_PORT + 1)),
+            mBobAddress(LoopbackTransport::LoopbackPeer(mAliceAddress))
+        {}
+        ~MessagingContextData() { EXPECT_FALSE(mInitialized); }
+
+        bool mInitializeNodes = true;
+        bool mInitialized;
```

Comment:
you initialize a lot of other things as `= ...`, however this one you initialize in the constructor. Why?

How about ` = false` on this one and also initialize all the pointers below to nullptr to have some initial state?

Reply:
Done.  Again, this was just my reluctance to change how the previous author had done things.  But that's not a good excuse.